hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py

```diff
@@ -51,21 +52,32 @@ def load(infile, uart):
     # - If XOR matches, host sends 0x6 as final ack
     # - board boots image after receiving ACK from host
 
-#    if len(msg) == 0:
-#        raise SystemExit("Read timed out, exiting")
-
-    print("Please reset board to enter ROM uart recovery")
+    som_detected = False
+    reset_triggered = False
+    now = datetime.datetime.now()
+    reset_delay_us = 250000
 
     while True:
+        elapsed = datetime.datetime.now() - now
+        if elapsed.seconds >= 15:
+            raise SystemExit("Failed to receive SOM, aborting")
+        if not som_detected and not reset_triggered:
+            if elapsed.microseconds >= reset_delay_us:
+                print("Triggering SWD reset...")
+                os.system("./reset.sh &")
```

Comment:
Doing this the script can only run `reset.sh` in the local directory, if I run it from repo root, eg, `./hw/bsp/dialog_da1469x-dk-pro/da1469x_serial.py`, it would not find the reset script. Not strictly required though.

Reply:
ah good point.  this really needs to be "reset.sh &" and the pathing must include the path to it.

Reply:
I am not a big run of `os.system` either, so I would use subprocess module. You could fix this with:

```
import os.path as path
import subprocess

reset = path.join(path.dirname(__file__), 'reset.sh')
subprocess.call(reset.split())
```

Reply:
For some reason, the serial read from the device after the reset doesn't receive expected data. I addressed all other comments except for switching os.system -> subprocess.call()